### PR TITLE
Add new script openmc-report-benchmarks

### PR DESCRIPTION
**Pull request by johvincau**
This pull request adds an automated LaTeX report production capability with a new script `openmc-report-benchmarks`. It takes one result file produced by `openmc-run-benchmarks` and produces a .tex file containing a table comparing the calculated criticality values and standard deviations to experimental benchmark values.

### CODE REVIEW

File: benchmarking/report.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    # Applying matching as needed (DEPRECATED)
+    if match is not None:
+        cond = index.map(lambda x: fnmatch(x, match))
+        index = index[cond]
+
+    # Custom Table Caption
+    caption = '\\caption{\\label{tab:1} Criticality (' + labels[0] + ') Benchmark Results}\\\\'
+    # Define Table Entry
+    table = [
+            'Table \\ref{tab:1} uses (nuclear data set info here) to evaluate ICSBEP benchmarks.',
```
Comment (shimwell) [inline benchmarking/report.py]:
```suggestion
            'Table \\ref{tab:1} uses (nuclear data set info here) to evaluate ICSBEP benchmarks.',
```
Perhaps a separate topic but if hdf5 nuclear data files had some tags in them so we could get the library name and the version from the hdf5 files automatically.

I guess we could include the version of openmc used automatically. With ```openmc.__version__```

Comment (johvincau) [reply benchmarking/report.py]: Thanks for taking a look. I'll add the automatic `openmc.__version__` that you suggested. As for the nuclear data files, I was trying to imitate the style of [this LANL report on MCNP](https://mcnp.lanl.gov/pdf_files/TechReport_2022_LANL_LA-UR-22-32951Rev.1_JoseyClarkEtAl.pdf), section 4.2 (image below). I'm not sure if there is a tag in the hdf5 files, but that would be a nice thing to add!

image

**Review by paulromano — CHANGES_REQUESTED** [review]
Thanks for this @johvincau. The table looks great! I made a [few style fixes](https://github.com/openmc-dev/validation/pull/8/commits/3cc7913c9dbe6ad9ba41ae1b78761ea9213c31b9) and pushed to your branch. Only other suggestion I have is that I think you could replace your `num_format` function with use of the `\num` command from the `siunitx` package, which will achieve the same spacing between thousands.

**Comment by johvincau** [comment]
Appreciate the style fixes! I just pushed a version which gets rid of `num_format` and adds `siunitx` to do the same thing, just as suggested.

**Review by paulromano — APPROVED** [review]
Thanks @johvincau!

**Pull request merged**